prime-angular-frontend/src/app/shared/components/forms/site-information-form/site-information-form.component.html

```diff
@@ -57,7 +57,7 @@
           <mat-label>Site ID</mat-label>
           <input matInput
                  formControlName="pec" />
-          <mat-error *ngIf="pec.hasError('required')">
+          <mat-error *ngIf="pec.hasError('required') || activeBeforeReg.checked">
```

Comment:
With this being a reactive form you will want to use a FormGroup validator apply validation to the PEC when activeBeforeReg is checked.

Reply:
As far as I can tell PEC is already always required